feat: remove flag icons (#13313)
shopware/platform: RELEASE_INFO-6.7.md

### Removal of old `changelog` handling
As we changed how we process and generate changelogs the "old" changelog files are no longer needed.
Therefore, we removed all the internal code used to generate and validate them.
The whole `Shopware\Core\Framework\Changelog` namespace was removed. The code is not needed anymore, you should adjust the `RELEASE_INFO` and `UPGRADE` files manually instead.

## Administration

## Storefront

> ### Language selector now displays language with territory instead of flags

Reviewer: ```suggestion
### Language selector will display language with territory instead of flags
```

> The language selector in the storefront will display the language name with its territory in the dropdown menu (e.g. "Deutsch (Deutschland)", "English (United Kingdom)"). The CSS for flag icons is deprecated and will be removed with Shopware 6.8.0.0.

Reviewer: ```suggestion
The language selector in the storefront will display the language name with its territory in the dropdown menu (e.g. "Deutsch (Deutschland)", "English (United Kingdom)") with the next major version.
The CSS for flag icons is deprecated and will be removed with Shopware 6.8.0.0.
```

> When the `v6.8.0.0` feature flag is active, the language selector in `src/Storefront/Resources/views/storefront/layout/header/actions/language-widget.html.twig` now displays language names with their territory in the dropdown (e.g., "Deutsch (Deutschland)", "English (United Kingdom)") instead of just the language name.

Reviewer: I would not speak of the feature flag, but more like that: "with the next major version..."
"the css flags are deprecated and will be removed with..." is also missing.

### Removal of old `changelog` handling
As we changed how we process and generate changelogs the "old" changelog files are no longer needed.
Therefore, we removed all the internal code used to generate and validate them.
The whole `Shopware\Core\Framework\Changelog` namespace was removed. The code is not needed anymore, you should adjust the `RELEASE_INFO` and `UPGRADE` files manually instead.

## Administration

## Storefront

### Language selector will displays language with territory instead of flags
The language selector in the storefront will display the language name with its territory in the dropdown menu (e.g. "Deutsch (Deutschland)", "English (United Kingdom)") with the next major version.
The CSS for flag icons is deprecated and will be removed with Shopware 6.8.0.0.

> New extensible Twig blocks `layout_header_actions_language_widget_content_inner` and `layout_header_actions_languages_widget_form_items_flag_inner` have been added to allow custom flag implementations.

Reviewer: ```suggestion
### Language selector will display language with territory instead of flags

The language selector in the storefront will display the language name with its territory in the dropdown menu (e.g. "Deutsch (Deutschland)", "English (United Kingdom)") from the next major version on.
The CSS for flag icons is deprecated and will then be removed.

New extensible Twig blocks `layout_header_actions_language_widget_content_inner` and `layout_header_actions_languages_widget_form_items_flag_inner` have been added to allow custom flag implementations.
```

## Storefront

### Language selector will display language with territory instead of flags

> The language selector in the storefront will display the language name with its territory in the dropdown menu (e.g. "Deutsch (Deutschland)", "English (United Kingdom)") from the next major version on.

Reviewer: I just had a small discussion with Max on another PR. 

as this is only documenting a breaking change, it needs to be in the upgrade guide only. the new twig blocks are okay here.
